File: webapp/rust/core/src/models/reservation_slot.rs
use fake::Dummy;

#[derive(Debug, Dummy)]
pub struct ReservationSlot {
    #[allow(unused)]
    pub id: ReservationSlotId,
    pub slot: i64,
    pub start_at: i64,
    pub end_at: i64,
}

kubetsu::define_id!(
    #[derive(toasty::Embed)]
    pub struct ReservationSlotId(i64);
);
kubetsu_serde::impl_serde!(ReservationSlotId(i64));
kubetsu_fake::impl_fake!(ReservationSlotId(i64));

File: webapp/go/domain/model/reservation_slot.go
package model

type ReservationSlotID = ID[ReservationSlotModel]

type ReservationSlotModel struct {
	ID      ReservationSlotID `db:"id"`
	Slot    int64             `db:"slot"`
	StartAt int64             `db:"start_at"`
	EndAt   int64             `db:"end_at"`
}
